src/tools/schema2kotlin.ts

```diff
@@ -339,11 +339,9 @@ ${this.opts.wasm ? `
     companion object : ${this.prefixTypeForRuntime('EntitySpec')}<${name}> {
         ${this.opts.wasm ? '' : `
         override val SCHEMA = ${leftPad(this.createSchema(schemaHash), 8, true)}.also { SchemaRegistry.register(it) }`}
-
-        override fun create() = ${name}()
         ${!this.opts.wasm ? `
         // TODO: only handles singletons for now
-        override fun deserialize(data: RawEntity) = create().apply { deserialize(data) }` : `
+        override fun deserialize(data: RawEntity) = ${name}().apply { deserialize(data) }` : `
```

Comment:
I see in arcs.core.entity.Entity and arcs.sdk.wasm.WasmEntity that `deserialize` returns an instance of the Entity, does it also modify the `this`?   I suspect the `apply` here returns the empty result of `${name}()`. (not new to this PR, but has likely been here a while)

Reply:
I was very confused here for a bit as well. 

The `deserialize` method that's invoked is the one in `EntityBase`, which mutates the instance and returns `Unit`.